### PR TITLE
Use ImplPtr, fix rule of 0

### PR DESCRIPTION
**Pull request by scpeters**
# 🎉 New feature

Use gz/utils/ImplPtr and fix rule of 0 violations

## Summary

This switches from raw data pointers to `gz::utils::ImplPtr` for `SpeedLimiter` and `MecanumOdometry` and deletes the destructor definitions to satisfy the rule of 0 for these classes.

Also remove some unneeded forward `class` declarations.

## Test it

Build and run unit tests.

## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [X] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [X] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by ahcorde — APPROVED** [review]

File: include/gz/math/MecanumDriveOdometry.hh [hunk 1]
```diff
@@ -18,20 +18,18 @@
...
 namespace gz
 {
   namespace math
   {
     // Inline bracket to help doxygen filtering.
     inline namespace GZ_MATH_VERSION_NAMESPACE {
-    // Forward declarations.
-    class MecanumDriveOdometryPrivate;
-
+    //
```
Comment (mjcarroll) [inline include/gz/math/MecanumDriveOdometry.hh:32]: delete

File: include/gz/math/SpeedLimiter.hh [hunk 2]
```diff
@@ -19,29 +19,24 @@
...
 namespace gz
 {
 namespace math
 {
 // Inline bracket to help doxygen filtering.
 inline namespace GZ_MATH_VERSION_NAMESPACE {
-  // Forward declaration.
-  class SpeedLimiterPrivate;
-
+  //
```
Comment (mjcarroll) [inline include/gz/math/SpeedLimiter.hh:32]: delete

File: include/gz/math/MecanumDriveOdometry.hh [hunk 3]
```diff
@@ -134,18 +129,7 @@ namespace gz
...
-// std::unique_ptr
-#pragma warning(push)
-#pragma warning(disable: 4251)
-#endif
-      /// \brief Private data pointer.
-      private: std::unique_ptr<MecanumDriveOdometryPrivate> dataPtr;
-#ifdef _WIN32
-#pragma warning(pop)
-#endif
+      GZ_UTILS_IMPL_PTR(dataPtr)
```
Comment (mjcarroll) [inline include/gz/math/MecanumDriveOdometry.hh:132]:
```suggestion
      /// \brief Private data pointer.
      GZ_UTILS_IMPL_PTR(dataPtr)
```

Here and elsewhere

Comment (scpeters) [reply include/gz/math/MecanumDriveOdometry.hh:132]: https://github.com/gazebosim/gz-math/pull/608/commits/78a90662c12bcf00e340e3477184caeef656738a

File: include/gz/math/SpeedLimiter.hh (hunk 2 above)
Comment (scpeters) [reply include/gz/math/SpeedLimiter.hh]: https://github.com/gazebosim/gz-math/pull/608/commits/78a90662c12bcf00e340e3477184caeef656738a

File: include/gz/math/MecanumDriveOdometry.hh (hunk 1 above)
Comment (scpeters) [reply include/gz/math/MecanumDriveOdometry.hh]: https://github.com/gazebosim/gz-math/pull/608/commits/78a90662c12bcf00e340e3477184caeef656738a

**Review by adityapande-1995 — APPROVED** [review]

**Pull request merged**